logdna/client_test.go

```diff
@@ -0,0 +1,91 @@
+package logdna
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestClient_View(t *testing.T) {
+	body := make(map[string](interface{}))
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		json.NewDecoder(r.Body).Decode(&body)
+	}))
+	defer ts.Close()
+
+	payload := ViewPayload{Name: "test", Query: "test"}
+	client := Client{ServiceKey: servicekey, HTTPClient: &http.Client{Timeout: 15 * time.Second}}
```

Comment:
currently the test server isn't being used at all, afaict

instead of passing a brand new `&http.Client` this should be something like `HTTPClient: &ts.Client()`:

https://golang.org/pkg/net/http/httptest/#Server.Client